agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -502,6 +507,19 @@ bool monitor_thread::update_sta_stats()
         auto vap_node   = mon_db.vap_get_by_id(sta_node->get_vap_id());
         auto &sta_stats = sta_node->get_stats();
 
+        // Skip stations that were already updated in the current cycle
+        if (sta_stats.last_update_time > m_sta_stats_polling_start_timestamp) {
+            continue;
+        }
+
+        if (std::chrono::steady_clock::now() > awake_timeout()) {
+            // If we haven't finish to iterate on all stations, skip one time on the select timeout
+            // so the select will not be stuck on full select timeout and the thread will be able to
+            // finish this operation quickly.
```

Comment:
IMO it's better to move this comment above the `if` condition, but maybe it's just my personal preference :)